### ISSUE TITLE
Omission of the presentation_definition_uri is the Browser API extension

### ISSUE THREAD

**Issue by Sakurann** [openid/OpenID4VP#182]
The omission of the presentation_definition_uri is understandable in that the platform needs to know which credentials are being requested in order to choose suitable wallets, and using PDURI instead would necessitate it calling out to a policy server to fetch the PD. However, the negative implication of this is that the trust framework will need to specify the entire PD when it indicates which credentials the RP is trusted to retrieve. Prior to this limitation, the trust framework only needed to hold the PD URI for the RP and the wallet could compare that with the PD URI provided by the RP in the request. One resolution of this maybe to allow the RP to specify both the PD and the PD URI. The platform will use the PD to choose the wallet,  and the wallet can choose to ignore the PD when determining if the RP is trustworthy, and to only use the PD URI which is confirmed by the trust framework to be the trustworthy one.

_Originally posted by @David-Chadwick in https://github.com/openid/OpenID4VP/pull/155#discussion_r1579730768_

**Comment by jogu** [comment]
I don't think it would be a good idea to create a situation where a different presentation definition is used for the user to select the credential to be returned than is used to actually return the credential.

I have to say I don't understand the model being described here where the trust framework apparently supplies the presentation definition. That would seem to prevent the RP from customising the presentation definition such that only the claims the verifier actually needs are returned? I've always assumed a model where the wallet validates that the verifier is entitled to receive the requested credential/claims rather than that entitlement being communicated via the presentation definition.

**Comment by David-Chadwick** [comment]
>I don't think it would be a good idea to create a situation where a different presentation definition is used for the user to select the credential to be returned than is used to actually return the credential.

that is not what is being suggested. The presentation_defintion_uri is used for both. The presentation_definition is only used by the browser/OS to choose the wallet that might contain the correct credentials

**Comment by David-Chadwick** [comment]
> I have to say I don't understand the model being described here where the trust framework apparently supplies the presentation definition. That would seem to prevent the RP from customising the presentation definition such that only the claims the verifier actually needs are returned? I've always assumed a model where the wallet validates that the verifier is entitled to receive the requested credential/claims rather than that entitlement being communicated via the presentation definition.

This is privacy issue and not an entitlement issue. Specifically it is to support the GDPR requirement of data minimisation. The RP should only request the minimal amount of information necessary for the transaction in hand. The trust framework lists the data that is needed for the transaction by pointing to the presentation definition. The user does not know if the RP is requesting more data than is actually needed, since the RP may be entitled to receive far more than what is actually required. Your comment seems to suggest that the RP is trusted to only request the minimal PII, whereas we all know that in practice RPs try to collect the most that they are entitled to.

**Comment by jogu** [comment]
> The presentation_definition is only used by the browser/OS to choose the wallet that might contain the correct credentials

That's not how the browser API works. The user selects a credential before the wallet is invoked. The outcome with the Browser API is what I said - a different presentation used for credential selection vs delivery.

I think we are talking about different use cases; I'm thinking more about the cases where the verifier is regulated and has to be approved to receive credentials, I think you're thinking about unregulated cases?

Is there more details somewhere about how the kind of trust framework you're describing works?

**Comment by David-Chadwick** [comment]
> The user selects a credential before the wallet is invoked.

In which case the user never needs to interact with the wallet. The browser API can call the wallet to request the selected credential, and it can return the requested credential directly to the browser. The wallet now becomes a dumb file store.

**Comment by David-Chadwick** [comment]
> I think we are talking about different use cases; I'm thinking more about the cases where the verifier is regulated and has to be approved to receive credentials, I think you're thinking about unregulated cases?

Not so. In our implementation of TRAIN, the trusted administrator (TA) vets the RP, and checks that the requested PII is needed for the transaction. The TA then enters the RP's details into the ETSI trust list (suitably modified to support VCs and VPs). Even though the RP has been vetted, there is no guarantee that it will act in a trustworthy manner. **The user/wallet trusts the TA, not the RP. This is the trust model.** So when the wallet receives the client id and presentation request uri it calls the TRAIN infrastructure which indicates that the RP is trusted, and that for the requested service the presentation request uri should be X. The wallet checks that X was indeed the presentation request uri provided by the RP i.e. checking that the RP is not now asking for more PII than is necessary.

If your trust model is that the RP is trusted (what you call regulated), then you do not need a trust infrastructure, you simply need a list of trusted RPs because your assumption is that the RP will always act in a trustworthy manner. There is no need to check what PII it requests.

**Comment by jogu** [comment]
> If your trust model is that the RP is trusted (what you call regulated), then you do not need a trust infrastructure, you simply need a list of trusted RPs because your assumption is that the RP will always act in a trustworthy manner. There is no need to check what PII it requests.

I don't think that automatically follows. It can very much still be sensible to limit the data a verifier can access to just the fields it has been approved to received, whilst also recognising that it is under a legal obligation to only request what it needs.

> In which case the user never needs to interact with the wallet.

The wallet usually still needs to deal with informing the user exactly which fields will be returned and who to - i.e. it is still expected that consent for data sharing will occur in the wallet.

I think there's a lot of overlap of this issue and issue https://github.com/openid/OpenID4VP/issues/189, but issue 189 should work fine with the browser architecture - do you think issue 189 solves covers the same requirement are you expressing here?

**Comment by David-Chadwick** [comment]
`It can very much still be sensible to limit the data a verifier can access to just the fields it has been approved to received, `
This sounds to me that the RP is NOT trustworthy in your model. Which is the model that we implemented in  TRAIN, where the trust infrastructure tells the wallet what the RP is trusted to ask for and the wallet checks that the RP's request conforms to this.

`do you think issue 189 solves covers the same requirement are you expressing here?`

It depends who is responsible for the RP's metadata. If the RP writes its own metadata, then clearly this is not trusted. But if the RP's metadata is provided by a trust infrastructure (which I think 189 implies), then it can be trusted and then it is very similar to the implementation that we did with TRAIN (we did not provide presentation definition IDs, but rather presentation definition URLs because all our presentation definitions were held in publicly accessible policy databases from where they could be downloaded). If the trust infrastructure only provides PD IDs, what mechanism is used to bind the ID to the PD, to ensure that the contents of the PD have not changed?

**Comment by jogu** [comment]
Thanks David, I see what you're saying there. I think that is the same root problem as #189 but I get your point that the proposed `presentation_definitions_supported` client metadata can't be self-asserted.

I have also correct my earlier comment - I no longer believe that #189 works with the browser API, so we currently have no proposals on the table that work with the browser API.

I think my proposal at the moment is to open a new issue that states the problem but doesn't pre-suppose any solution and close both this issue and #189 in favour of that issue.

**Comment by David-Chadwick** [comment]
Can you open the new issue first, and then if we agree that it adequately addresses both this issue and #189 then they both can be closed. Thanks

**Comment by jogu** [comment]
Thanks @David-Chadwick - I opened https://github.com/openid/OpenID4VP/issues/232, please let me know if I missed anything and I can edit it. If it's okay I suggest we close this issue and continue work on #232.

**Comment by David-Chadwick** [comment]
Yes #232 is fine. You can close the current issue now.

**Comment by jogu** [comment]
Thanks David!
